src/plugins/drawing-tool/components/drawing-layer.tsx

```diff
@@ -172,6 +172,10 @@ export class DrawingLayerView extends React.Component<DrawingLayerViewProps, Dra
   public handleObjectClick = (e: MouseEvent|React.MouseEvent<any>, obj: DrawingObjectType) => {
     if (!this.props.readOnly && this.currentTool) {
       this.currentTool.handleObjectClick(e, obj);
+      if (this.state.selectedObjects.length> 1 && !(e.shiftKey || e.metaKey)){
+        const filteredArray = this.state.selectedObjects.filter(selObj=> selObj === obj);
+        this.setState(prevState => ({selectedObjects: filteredArray}));
+      }
```

Comment:
I'd think you should keep this more simple and just do:

```suggestion
      if (!(e.shiftKey || e.metaKey)){
        this.setState(prevState => ({selectedObjects: [selObj}));
      }
```
This would make sure the object being clicked on becomes the only selected object. With the filter approach if the object isn't already in the selectedObjects then clicking on it won't select it. But I haven't really groked how all of your changes work together.

Also rather than changing the selection using setState directly you should use `setSelectedObjects` because this updates the state in the react component and the model.

Reply:
Looking at this further. This logic should be moved into `SelectionDrawingTool.handleObjectClick` that is what is currently managing the selection on click. It does it by calling `setSelectedObjects`. 
It has access to the mouse event so it can decide if the modifier key is pressed.